app/features/user/auth.ts

```diff
@@ -43,14 +51,67 @@ type AuthState = {
     }
 );
 
+const userRepository = new UserRepository(boardwalkDb);
+
+const useUpsertBoardwalkUser = (openSecretUserData?: OpenSecretUser | null) => {
+  const [error, setError] = useState<unknown | null>(null);
+  const [boardwalkUserData, setBoardwalkUserData] = useState<{
+    loading: boolean;
+    user: BoardwalDbkUser | null;
+  }>({ loading: !!openSecretUserData, user: null });
+
+  useEffect(() => {
+    const upsertBoardwalkUser = async () => {
+      if (!openSecretUserData) {
+        return;
+      }
+
+      try {
+        const boardwalkUser = await userRepository.upsert(openSecretUserData);
+        setBoardwalkUserData({ loading: false, user: boardwalkUser });
+      } catch (e) {
+        setError(e);
+      }
+    };
+    upsertBoardwalkUser();
+  }, [openSecretUserData]);
+
+  if (error) {
+    throw error;
+  }
+
+  return boardwalkUserData;
+};
+
+const useSetSupabseSession = (openSecretUserData?: OpenSecretUser | null) => {
+  useEffect(() => {
+    if (!openSecretUserData) {
+      return;
+    }
+    supabaseSessionStore
+      .getState()
+      .setJwtPayload({ sub: openSecretUserData.id });
+  }, [openSecretUserData]);
+};
+
 export const useAuthState = (): AuthState => {
   const {
-    auth: { user: userResponse, loading },
+    auth: { loading: loadingOpenSecretUser, user: openSecretUserResponse },
   } = useOpenSecret();
-  const userData = userResponse?.user;
+  const openSecretUserData = openSecretUserResponse?.user;
+
+  useSetSupabseSession(openSecretUserData);
+
+  const { loading: loadingBoardwalkUser, user: boardwalkUserData } =
+    useUpsertBoardwalkUser(openSecretUserData);
```

Comment:
I was thinking how to handle this. Initially I thought I will insert user data to postgres when user is created immediately after it was created in open secret but then there was additional complexity in handling the case where open secret registration succeeds and storing to postgres fails.
Because of that I decided to just do it every time when open secret user is loaded or changed. This means even if storing to postgres fails another page load will try it again.